app/models/category_list.rb

```diff
@@ -156,7 +156,11 @@ def find_categories
     notification_levels = CategoryUser.notification_levels_for(@guardian.user)
     default_notification_level = CategoryUser.default_notification_level
 
-    if @options[:parent_category_id].blank?
+    if SiteSetting.lazy_load_categories
+      @categories.each do |c|
+        c.subcategory_ids = Category.where(parent_category_id: c.id).pluck(:id)
```

Comment:
This is going to cause an N+1 queries problem because a DB query is being triggered for each iteration in the loop.

Reply:
Oh yes. I implemented this quickly for the prototype and forgot about it. I fixed it.